cql3/query_processor.cc

```diff
@@ -119,62 +127,134 @@ query_processor::query_processor(service::storage_proxy& proxy, database& db, qu
+                            }),
 
                     sm::make_derive(
                             "updates",
-                            _cql_stats.statements[size_t(statement_type::UPDATE)],
-                            sm::description("Counts a total number of CQL UPDATE requests.")),
+                            sm::description("Counts the total number of CQL UPDATE requests."),
+                            [this] {
+                                return _cql_stats.query_cnt(source_selector::USER, ks_selector::SYSTEM, stm::statement_type::UPDATE)
+                                        + _cql_stats.query_cnt(source_selector::USER, ks_selector::NONSYSTEM, stm::statement_type::UPDATE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::SYSTEM, stm::statement_type::UPDATE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::NONSYSTEM, stm::statement_type::UPDATE);
+                            }),
 
                     sm::make_derive(
                             "deletes",
-                            _cql_stats.statements[size_t(statement_type::DELETE)],
-                            sm::description("Counts a total number of CQL DELETE requests.")),
+                            sm::description("Counts the total number of CQL DELETE requests."),
+                            [this] {
+                                return _cql_stats.query_cnt(source_selector::USER, ks_selector::SYSTEM, stm::statement_type::DELETE)
+                                        + _cql_stats.query_cnt(source_selector::USER, ks_selector::NONSYSTEM, stm::statement_type::DELETE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::SYSTEM, stm::statement_type::DELETE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::NONSYSTEM, stm::statement_type::DELETE);
+                            }),
+
+                    sm::make_derive(
+                            "per_ks_reads",
+                            _cql_stats.query_cnt(source_selector::USER, ks_selector::SYSTEM, stm::statement_type::SELECT),
+                            sm::description("Counts the number of CQL SELECT requests executed on particular keyspaces. "
+                                            "Label `who' indicates where the reqs come from (clients or DB internals)"),
+                            {user_who_label_instance, system_ks_label_instance}),
+
+                    sm::make_derive(
+                            "per_ks_reads",
+                            _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::SYSTEM, stm::statement_type::SELECT),
+                            sm::description(),
+                            {internal_who_label_instance, system_ks_label_instance}),
+
+                    sm::make_derive(
+                            "per_ks_inserts",
```

Comment:
if you are adding the per_ks please add it as a suffix, it would make searching easier

Reply:
Done.
Apparently Github didn't reload this PR after I force pushed, but it should be able to be cleanly merged (even by ff-only). Unless you have more comments, of course.